Backend/src/main/java/com/SYSC4806_Group13/SYSC4806_Project/Controllers/ListingRestController.java

```diff
@@ -145,16 +146,19 @@ public Map<String, Object> putListing(@RequestBody Map<String, Object> payload)
             throw new MissingAttributeException("Attribute 'price' must be formatted in decimal as a string (ex. '1.5')");
         }
 
-        repo.save(listing);
+        listingRepo.save(listing);
+        CartItemRestController.updateListingInventory(listing.getListingId(), cartItemRepo);
         return map;
     }
 
     @DeleteMapping("/listings/{id}")
     public Listing deleteListing(@PathVariable String id) {
-        Optional<Listing> listing = repo.findById(Long.parseLong(id));
+        Optional<Listing> listing = listingRepo.findById(Long.parseLong(id));
         if (listing.isPresent()) {
             listing.get().setActive(false);
-            repo.save(listing.get());
+            listing.get().setInventory(0);
```

Comment:
Are we setting this to 0 just to force all associated cart items to get deleted?

Reply:
Yes zeroing the inventory upon deletion is just for updating the related cart items after deletion. 
Note that there is currently nowhere in the frontend that uses this endpoint as of right now though

Reply:
I feel that the extensibility in the newly renamed `updateAssociatedCartItems_listingInventoryUpdated` static method would be sufficient to handle the only deletion case as that use case is partially covered by zeroing an active listing's inventory already and I'd prefer that the code not be repeated